library/include/rocwmma/internal/types_ext.hpp

```diff
@@ -68,43 +100,84 @@ namespace rocwmma
 
     __host__ inline hfloat16_t operator+(const hfloat16_t& x, const hfloat16_t& y)
```

Comment:
Originally, operators for __half are not defined on host only.
But the new case where we have __ HIP_NO_HALF_OPERATORS __, we would now have to define both host AND device.
I recommend something like this:
```
#if !defined(__HIP_NO_HALF_OPERATORS__)
#define ROCWMMA_HALF_OP_ATTR ROCWMMA_HOST
#else 
#define ROCWMMA_HALF_OP_ATTR ROCWMMA_HOST_DEVICE
#endif
```
Then use the ROCWMMA_HALF_OP_ATTR for the operator functions

Reply:
Add: This also needs to apply to ALL operators, even the operator==, != and - above